### PR TITLE
Re-instate plugin registration checks for adapters.

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: pkg/adapters/grpc/document_grpc.go [hunk 1]
```diff
@@ -30,7 +32,19 @@ type DocumentServiceServer struct {
 	documentPlugin document.DocumentService
 }
 
+func (s *DocumentServiceServer) checkPluginRegistered() error {
+	if s.documentPlugin == nil {
+		return status.Errorf(codes.Unimplemented, "Document plugin not registered")
+	}
+
```
Comment (medgar-nitric) [inline pkg/adapters/grpc/document_grpc.go:39]: Could be worth creating an error helper function in the package ```utils.go``` to help standardize these messages:

```
func PluginNotRegisteredError(name string) error {
	return status.Errorf(codes.Unimplemented, "%s plugin not registered", name)
}
```

File: pkg/adapters/grpc/queue_grpc.go [hunk 2]
```diff
@@ -28,7 +30,19 @@ type QueueServiceServer struct {
 	plugin queue.QueueService
 }
 
+func (s *QueueServiceServer) checkPluginRegistered() error {
+	if s.plugin == nil {
+		return status.Errorf(codes.Unimplemented, "Secret plugin not registered")
```
Comment (medgar-nitric) [inline pkg/adapters/grpc/queue_grpc.go:35]: Please change error message to ```Queue plugin not registered```

File: pkg/adapters/grpc/storage_grpc.go [hunk 3]
```diff
@@ -27,7 +29,19 @@ type StorageServiceServer struct {
 	storagePlugin storage.StorageService
 }
 
+func (s *StorageServiceServer) checkPluginRegistered() error {
+	if s.storagePlugin == nil {
+		return status.Errorf(codes.Unimplemented, "Secret plugin not registered")
```
Comment (medgar-nitric) [inline pkg/adapters/grpc/storage_grpc.go:34]: Please change error message to ```Storage plugin not registered```

**Review by medgar-nitric — CHANGES_REQUESTED** [review]

File: pkg/adapters/grpc/document_grpc.go (hunk 1 above)
Comment (tjholm) [reply pkg/adapters/grpc/document_grpc.go:37]: Done

File: pkg/adapters/grpc/storage_grpc.go (hunk 3 above)
Comment (tjholm) [reply pkg/adapters/grpc/storage_grpc.go]: Done

File: pkg/adapters/grpc/queue_grpc.go (hunk 2 above)
Comment (tjholm) [reply pkg/adapters/grpc/queue_grpc.go]: Done

File: pkg/adapters/grpc/event_grpc.go [hunk 4]
```diff
@@ -28,7 +30,19 @@ type EventServiceServer struct {
 	eventPlugin events.EventService
 }
 
+func (s *EventServiceServer) checkPluginRegistered() error {
+	if s.eventPlugin == nil {
+		return status.Errorf(codes.Unimplemented, "Event plugin not registered")
```
Comment (medgar-nitric) [inline pkg/adapters/grpc/event_grpc.go:35]: Can you use the new error function to here.

**Review by medgar-nitric — CHANGES_REQUESTED** [review]

File: pkg/adapters/grpc/event_grpc.go (hunk 4 above)
Comment (tjholm) [reply pkg/adapters/grpc/event_grpc.go]: done.

**Review by medgar-nitric — APPROVED** [review]

**Review by jyecusch — APPROVED** [review]

**Pull request merged**